AEP: update long-term correction and remove IAV for now

nbodini opened the pull request:
This PR addresses the issue described in #99 
It also adds/fixes a couple minor things:
- "era5" is now one of the default reanal_products in __init__
- mse and r2 are now saved in sim_results

Note that IAV has been temporarily turned off because of the issues described in #86

RHammond2 reviewed (changes requested):
From a software perspective, this looks good, and I'd just update the one docstring so that it matches what is being returned with this update.

nbodini commented on operational_analysis/methods/plant_analysis.py:
```diff
@@ -873,33 +867,25 @@ def sample_long_term_reanalysis(self):
-
         Returns:
            :obj:`pandas.DataFrame`: the windiness-corrected or 'long-term' annualized monthly/daily wind speeds
            :obj:`pandas.DataFrame`: Long-term wind speed stdev, from which IAV will be derived
```
Good catch!

ejsimley commented on operational_analysis/methods/plant_analysis.py:
```diff
@@ -844,14 +835,14 @@ def run_AEP_monte_carlo(self):
             if self.time_resolution == 'M': # Undo normalization to 30-day months
-                gross_lt = gross_lt*self.num_days_lt/30 
+                gross_lt = gross_lt*np.tile(self.num_days_lt,self._run.num_years_windiness)/30 
                 gross_por = gross_por*self.num_days_lt/30
```
As we were discussing in PR #105 we should update how gross_por is calculated, so it uses the new LT correction method too, and uses inputs from _reanalysis_por instead of _reanalysis_por_avg.

nbodini replied:
@ejsimley I agree it makes sense to fix this issue here. I've taken a first look and it's a bit more complicated than what I was expecting, because we need to be careful about seasonality impacts: if we calculate gross_por simply using _reanalysis_por, we won't necessarily get a gross energy estimate that is equally distributed throughout the year (e.g. if the POR goes from December 2016 to February 2020, winter would be weighted more in gross_por, and this should be taken into account when then calculating lt_por_ratio). Let me know if you can think of an easy way to solve this.
 
In the meantime, I've noticed an issue in how lt_por_ratio was being calculated, and I've fixed it.

nbodini replied:
@ejsimley, I've been thinking a bit more about this, and still couldn't find an easy fix. I guess one option could be trimming the POR data and calculating gross_por with the new LT approach using n*12 months. Using the old LT correction to calculate gross_por doesn't make much sense to me (and would lead to misleading results when using nonlinear ML models). The other option would be to just remove from the class the couple of lines to calculate gross_por (it's just used to calculate lt_por_ratio which is given as output of the class, not sure how important that is?), but that's not good either...

nbodini replied:
@ejsimley I think I was able to fix this! gross_por is now calculated using the new LT approach (i.e., the regression model is applied to the raw por data, without calculating any intermediate averages), and I've deleted _reanalysis_por_avg as it is not needed anymore with the new approach. To fix the seasonality issue described above, self.groupby_time_res is applied to gross_por to calculate an average gross energy value for each month (so 12 values total), and this "average" year is then used to calculate lt_por_ratio.

ejsimley replied:
I agree this incorporates the new LT approach while avoiding seasonality issues. Looks good to me!

ejsimley reviewed (commented):
The new LT correction method, and removing IAV for now, looks good to me. As we discussed, we still need to update how gross_por is calculated. I was debating if that should be a separate issue/PR, but I think we could just resolve that here. @nbodini, feel free to try to fix that, but I'm happy to take a look at it too. Let me know what you think!

RHammond2 reviewed (approved):
From my standpoint, this looks good with the updated docstrings.

RHammond2 commented:
@jordanperr have you seen an error like this before on the RTD build? The issue seems to be stemming from the usage of easy_install and building from an egg rather than wheel, but in the binder PR I don't have this issue and it was pre-3.0, so maybe that's another part of it. I haven't been able to find much on this issue other than [eggs are deprecated and we're supposed to use wheels](https://setuptools.readthedocs.io/en/latest/userguide/commands.html#bdist-egg-create-a-python-egg-for-the-project). Happy to keep digging if you don't have and leads on this.

```
Installed /home/docs/checkouts/readthedocs.org/user_builds/openoa/envs/102/lib/python3.7/site-packages/Shapely-1.7.1-py3.7-linux-x86_64.egg
Searching for pyproj
Reading https://pypi.org/simple/pyproj/
Downloading https://files.pythonhosted.org/packages/17/e5/3f5cdff3e955bcd768cdb0f4236f2d6e022aaa72f57caf7f4d5f552c88fc/pyproj-3.0.0.post1.tar.gz#sha256=a49581629cadd29e61fc061d153a4d62ff28b4063c71fe8ca881eeb98cd22017
Best match: pyproj 3.0.0.post1
Processing pyproj-3.0.0.post1.tar.gz
Writing /tmp/easy_install-frr172od/pyproj-3.0.0.post1/setup.cfg
Running pyproj-3.0.0.post1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-frr172od/pyproj-3.0.0.post1/egg-dist-tmp-eww3ihtg
error: Setup script exited with ERROR: Cython.Build.cythonize not found. Cython is required to build from a repo.
```

nbodini commented:
I have now added a couple of other minor edits still connected to the LT correction:
1) leap years are now better accounted for in the sample_long_term_reanalysis function, when the AEP calculation is performed at daily resolution. For this, self._calendar_samples at daily resolution has been now set to 365.25 (instead of 365).
2) I have fixed the issue (noted by Jordan in the comments of the class) of some abuse of the reset_index within the sample_long_term_reanalysis function. Now the output of that function keeps as index the time information (while before this info was being lost).

ejsimley reviewed (approved):
The latest changes look good and the examples are all running fine. OK with me to merge after updating the documentation where I left a comment. Perhaps Jordan wants to look at the changes too.

nbodini commented on operational_analysis/methods/plant_analysis.py:
```diff
@@ -873,33 +867,21 @@ def sample_long_term_reanalysis(self):
            (None)
-
         Returns:
            :obj:`pandas.DataFrame`: the windiness-corrected or 'long-term' annualized monthly/daily wind speeds
```
I've updated the documentation!

ejsimley commented:
@jordanperr Note that I created issue #123 related to AEP analysis failing when wind direction is added as a regression variable. But since the code fails in `apply_IAV` this PR (which removes IAV for now) should resolve the issue.

jordanperr commented:
Looks good to me. Thanks for your contribution, Nicola! I apologize for the lengthly review period.
